modules/manager/watchnamespaces.go

```diff
@@ -0,0 +1,25 @@
+package manager
+
+import (
+	"errors"
+	"strings"
+)
+
+// NewWatchNamespaces parses the watchNamespaces string and returns a slice
+// of trimmed namespace names.
+func NewWatchNamespaces(watchNamespaces string) ([]string, error) {
+	if watchNamespaces == "" {
+		return nil, nil
+	}
+
+	// Split the namespaces by comma and trim whitespace
+	namespaces := make([]string, 0)
+	for ns := range strings.SplitSeq(watchNamespaces, ",") {
```

Comment:
:+1: for using `SplitSeq` that can return early when error happens without making the whole string slice.